src/client/opamArg.mli

```diff
@@ -16,17 +16,54 @@ open Cmdliner
 
 (** {2 Helpers and argument constructors} *)
 
-val mk_flag: ?section:string -> string list -> string -> bool Term.t
+(** {3 CLI versioning} *)
+
+type flag_validity =
+  (* New flag since version [OpamCLIVersion.t] *)
+  | Valid_since of OpamCLIVersion.t
+  (* since * removal * instead
+     Option to remove, deprecated [since], to be removed at [removal],
+     and a hint to use another option or command [instead] *)
+  | Deprecated of
+      OpamCLIVersion.t (* since *)
+      * OpamCLIVersion.t (* removal *)
+      * string option (* use instead *)
```

Comment:
This naming makes a lot of sense to me, but maybe not for everyone :) If it is not clear, it is possible to renames it. It should be something easy to understand as  it is mandatory from this PR for every flag/option/subcommand handling.